src/main/java/com/commercetools/pspadapter/notification/processor/impl/PaymentSaleSimpleProcessorBase.java

```diff
@@ -20,11 +20,11 @@ public PaymentSaleSimpleProcessorBase(@Nonnull Gson gson) {
         super(gson);
     }
 
-    List<? extends UpdateAction<Payment>> createUpdateCtpTransactionActions(@Nonnull Payment ctpPayment,
-                                                                            @Nonnull Event event,
-                                                                            @Nonnull TransactionState transactionState) {
-        return findTransactionByTypeAndState(ctpPayment.getTransactions(), TransactionType.CHARGE, TransactionState.PENDING)
-                .map(txn -> singletonList(ChangeTransactionState.of(transactionState, txn.getId())))
+    List<? extends UpdateAction<Payment>> createUpdateCtpTransactionState(@Nonnull Payment ctpPayment,
+                                                                          @Nonnull TransactionState fromTransactionState,
+                                                                          @Nonnull TransactionState toTransactionState) {
+        return findTransactionByTypeAndState(ctpPayment.getTransactions(), TransactionType.CHARGE, fromTransactionState)
```

Comment:
let's wait for the response from the PP support how to identify transactions by ID

Reply:
Can we conclude we got an answer from PP support and I should use `resource.id` to match?

Reply:
yes, use it. You have to change notification handling to the next behavior:
  - if transaction is not found - add it as a new transaction to the payment
  - otherwise update previous transaction status/state (found by `resource.id`)

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/69/commits/2fdf5b7ffb502f2140dc3e77c1187a75ad5a3b3d